test/extensions/filters/http/rbac/rbac_filter_test.cc

```diff
@@ -51,38 +56,125 @@ class RoleBasedAccessControlFilterTest : public testing::Test {
     (*config.mutable_shadow_rules()->mutable_policies())["bar"] = shadow_policy;
     config.set_shadow_rules_stat_prefix("prefix_");
 
-    return std::make_shared<RoleBasedAccessControlFilterConfig>(
-        config, "test", store_, ProtobufMessage::getStrictValidationVisitor());
+    setupConfig(std::make_shared<RoleBasedAccessControlFilterConfig>(
+        config, "test", store_, context_, ProtobufMessage::getStrictValidationVisitor()));
   }
 
-  RoleBasedAccessControlFilterTest()
-      : config_(setupConfig(envoy::config::rbac::v3::RBAC::ALLOW)), filter_(config_) {}
+  void setupMatcher(envoy::config::rbac::v3::RBAC::Action rbac_action,
+                    envoy::config::rbac::v3::RBAC::Action on_no_match_rbac_action) {
+    envoy::extensions::filters::http::rbac::v3::RBAC config;
 
-  void SetUp() override {
-    config_ = setupConfig(envoy::config::rbac::v3::RBAC::ALLOW);
-    filter_ = RoleBasedAccessControlFilter(config_);
+    envoy::extensions::matching::common_inputs::network::v3::ServerNameInput server_name_input;
+    envoy::extensions::matching::common_inputs::network::v3::DestinationPortInput
+        destination_port_input;
+    envoy::type::matcher::v3::HttpRequestHeaderMatchInput header_input;
+    header_input.set_header_name(":path");
+
+    envoy::config::rbac::v3::Action action;
+    action.set_name("foo");
+    action.set_action(rbac_action);
+    envoy::config::rbac::v3::Action shadow_action;
+    shadow_action.set_name("bar");
+    shadow_action.set_action(rbac_action);
+    envoy::config::rbac::v3::Action on_no_match_action;
+    on_no_match_action.set_name("none");
+    on_no_match_action.set_action(on_no_match_rbac_action);
+
+    xds::type::matcher::v3::Matcher matcher;
+    {
```

Comment:
the new matcher API is much more complicated than the current RBAC API, it gets really hard to figure out the final content of the `matcher` that is being tested here when it is construed from a list of separate statements like below, could we instead just parse it from yaml so that it's much easier to understand the final content of the object and also what is actually being tested?

Reply:
Done.